avocado/core/task/runtime.py

```diff
@@ -114,35 +114,42 @@ def get_requirements_form_runnable(cls, runnable, status_server_uri=None,
                        sent to the destination job's status server and will
                        make into the job's results.
         :type job_id: str
-        :returns: RUntimeTasks of the requirements from runnable
+        :returns: RUntimeTasks of the dependencies from runnable
         :rtype: list
         """
-        if runnable.requirements is None:
+        if runnable.dependencies is None:
             return []
 
-        # creates the runnables for the requirements
-        requirements_runnables = RequirementsResolver.resolve(runnable)
-        requirements_runtime_tasks = []
-        # creates the tasks and runtime tasks for the requirements
-        for requirement_runnable in requirements_runnables:
-            name = '%s-%s' % (requirement_runnable.kind,
-                              requirement_runnable.kwargs.get('name'))
+        # creates the runnables for the dependencies
+        dependencies_runnables = DependenciesResolver.resolve(runnable)
+        dependencies_runtime_tasks = []
+        # creates the tasks and runtime tasks for the dependencies
+        for dependence_runnable in dependencies_runnables:
+            name = '%s-%s' % (dependence_runnable.kind,
+                              dependence_runnable.kwargs.get('name'))
             prefix = 0
             # the human UI works with TestID objects, so we need to
             # use it to name Task
             task_id = TestID(prefix, name)
-            # with --dry-run we don't want to run requirement
+            # with --dry-run we don't want to run dependencies
             if runnable.kind == 'dry-run':
-                requirement_runnable.kind = 'noop'
-            # creates the requirement task
-            requirement_task = Task(requirement_runnable,
-                                    identifier=task_id,
-                                    status_uris=status_server_uri,
-                                    category='requirement',
-                                    job_id=job_id)
-            requirements_runtime_tasks.append(cls(requirement_task))
+                dependence_runnable.kind = 'noop'
+            # creates the dependence task
+            dependence_task = Task(dependence_runnable,
+                                   identifier=task_id,
+                                   status_uris=status_server_uri,
+                                   category='dependence',
+                                   job_id=job_id)
+            dependencies_runtime_tasks.append(cls(dependence_task))
 
-        return requirements_runtime_tasks
+        return dependencies_runtime_tasks
+
+    def is_dependencies_finished(self):
+        for dependencie in self.dependencies:
+            if not dependencie.status or not ("FINISHED" in dependencie.status
+                                              or "FAILED" in dependencie.status):
```

Comment:
if you don't mind `dependencie` -> `dependency`

Reply:
In second thought, I think it's [dependence](https://en.wiktionary.org/wiki/dependence#English)) and the plural would be `dependences`.

Reply:
Actually I am not really sure what is the exact difference between `dependency` and `dependence`. But it looks like in our context the `dependency` is [better](https://ell.stackexchange.com/questions/41528/differences-between-dependence-and-dependency).